src/util.ts

```diff
@@ -419,9 +419,7 @@ export class Util {
       try {
         parsedHttpRespBody.body = JSON.parse(body);
       } catch (err) {
-        parsedHttpRespBody.err = new ApiError(
```

Comment:
What is the reason behind ditching this? It could very well be a problem, but I'd like to understand more, so we don't make the same mistake again.

Reply:
We discovered the API could return plain-text responses in certain 5xx errors.

Reply:
If it's only 500s, why not let it still be an "ApiError"?

Reply:
I think the problem is the error code is not propagated through when it's plain text.
```javascript
  describe.only('test', () => {
    nock('http://test.com')
      .get('/')
      .reply(503, "<html>Service unavailable.</html>");

    it('test', (done) => {
      util.makeRequest({uri: 'http://test.com'}, {}, (err: any, body: any, resp: any) => {
        // expect err.code to be 400, but got undefined
        assert.strictEqual(err.code, 503);
        done();
      });
    })
  });
```

Reply:
I see now. I think you're right, we can leave out the "Cannot parse as JSON" part. The only change I'd make is to not do anything in the "catch" handler, since `body` is already assigned at the top of the function.

Reply:
@bcoe also suggested up top that we could perhaps look at the content-type header from the response and keep throwing in the case where content-type is `application/json` and the response isn't parseable. WDYT?

Reply:
I think the only thing this block did was attempt to JSON.parse the body, then return the error body with the prefix "Could not parse..." if it failed. But, it already determines there's an error and creates an ApiError in the previous call to the "parseHttpRespMessage" function. So that original ApiError ends up being overwritten by this one, the "Could not parse..." one, which doesn't have as much information on it (the `code` property). So, I think I'm okay deleting it completely, like you have it now, since it wasn't ever really helping, as far as I can tell.